src/mlpack/core/util/param.hpp

```diff
@@ -1228,6 +1230,11 @@ using DatasetInfo = DatasetMapper<IncrementPolicy, std::string>;
       JOIN(io_option_dummy_object_in_, __COUNTER__) \
       (DEF, ID, DESC, ALIAS, #T, REQ, true, false, testName);
 
+  #define PARAM_IN_WITH_NAME(T, ID, DESC, ALIAS, NAME, DEF, REQ) \
+      static mlpack::util::Option<T> \
+      JOIN(io_option_dummy_object_in_, __COUNTER__) \
+      (DEF, ID, DESC, ALIAS, NAME, REQ, true, false, testName);
```

Comment:
Nice, this looks good!  To keep the code maintainable (and minimal), can you adapt `PARAM_COL()`, `PARAM_UCOL()`, etc. to be expressed in terms of this macro?